packages/types/src/oauth.ts

```diff
@@ -255,7 +251,28 @@ export type AuthenticateWithRedirectParams = {
    */
   redirectUrlComplete: string;
   /**
-   * Whether to continue (i.e. PATCH) an existing SignUp (if present) or create a new SignUp
+   * Whether to continue (i.e. PATCH) an existing SignUp (if present) or create a new SignUp.
    */
   continueSignUp?: boolean;
-};
+} & (
+  | {
+      /**
+       * One of the supported OAuth providers you can use to authenticate with, eg 'oauth_google'.
+       */
+      strategy: OAuthStrategy;
+      /**
+       * Email address not expected for OAuth case.
+       */
+      emailAddress: never;
+    }
+  | {
```

Comment:
🔧 Consider moving this types outside oauth.ts as they are not just about OAuth anymore.

Reply:
Moving outside of `oauth.ts`.